runtimes/alibi-explain/mlserver_alibi_explain/common.py

```diff
@@ -82,17 +101,49 @@ def import_and_get_class(class_path: str) -> type:
     return klass
 
 
-def to_v2_inference_request(input_data: Union[np.ndarray, List[str]]):
+def to_v2_inference_request(
+    input_data: Union[np.ndarray, List[str]],
+    metadata: Optional[MetadataModelResponse],
+) -> InferenceRequest:
+    """
+    Encode numpy payload to v2 protocol.
+
+    Note: We only fetch the first-input name and the list of outputs from the metadata
+    endpoint currently. We should consider wider reconciliation with data types etc.
+
+    Parameters
+    ----------
+    input_data
+       Numpy ndarray to encode
+    metadata
+       Extra metadata that can help encode the payload.
+    """
+
+    # MLServer does not really care about a correct input name!
+    input_name = _DEFAULT_INPUT_NAME
+    id_name = _DEFAULT_ID_NAME
+    outputs = []
+
+    if metadata is not None:
+        if metadata.inputs:
+            # we only support a big single input numpy
+            input_name = metadata.inputs[0].name
+        if metadata.outputs:
+            outputs = metadata.outputs
+
     # For List[str] (e.g. AnchorText), we use StringCodec for input
     input_payload_codec = StringCodec if type(input_data) == list else NumpyCodec
     v2_request = InferenceRequest(
+        id=id_name,
         parameters=Parameters(content_type=input_payload_codec.ContentType),
         # TODO: we probably need to tell alibi about the expected types to use
         # or even whether it is a probability of classes or targets etc
         inputs=[
             input_payload_codec.encode_request_input(  # type: ignore
-                name="predict", payload=input_data
+                name=input_name, payload=input_data
             )
         ],
+        # set outputs as empty list, this will return everything?
```

Comment:
I don't think that the protocol itself defines what should happen, so I guess that the behaviour will depend on the model and model runtime.

Reply:
actually this comment is not valid anymore as we get that from the metadata.